packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,294 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+const _ = require('lodash');
+const ArtifactGenerator = require('../../lib/artifact-generator');
+const debug = require('../../lib/debug')('repository-generator');
+const inspect = require('util').inspect;
+const path = require('path');
+const chalk = require('chalk');
+const utils = require('../../lib/utils');
+
+const SERVICE_VALUE_CONNECTOR = 'soap,rest';
+const KEY_VALUE_CONNECTOR = 'kv-';
+
+const KEY_VALUE_REPOSITORY = 'KeyValueRepository';
+const DEFAULT_CRUD_REPOSITORY = 'DefaultCrudRepository';
+
+const REPOSITORY_KV_TEMPLATE = 'repository-kv-template.ts.ejs';
+const REPOSITORY_CRUD_TEMPLATE = 'repository-crud-default-template.ts.ejs';
+
+const PROMPT_MESSAGE_MODEL =
+  'Select the model(s) you want to generate a repository';
+const PROMPT_MESSAGE_DATA_SOURCE = 'Please select the datasource';
+const PROMPT_MESSAGE_ID_TYPE = 'What is the type of your ID?';
+
+const ERROR_READING_FILE = 'Error reading file';
+const ERROR_NO_DATA_SOURCES_FOUND = 'No datasources found in';
+const ERROR_NO_MODELS_FOUND = 'No models found in';
+const ERROR_NO_MODEL_SELECTED = 'You did not select any model';
+
+module.exports = class RepositoryGenerator extends ArtifactGenerator {
+  // Note: arguments and options should be defined in the constructor.
+  constructor(args, opts) {
+    super(args, opts);
+
+    /** instance helper method isolated from the execution loop
+     * @connectorType: can be a single or a comma separated string list
+     */
+    this.isConnectorType = async function(connectorType, dataSourceClassName) {
```

Comment:
Helper methods should be defined as regular class methods and their name should start with `_`:

```ts
module.exports = class RepositoryGenerator extends ArtifactGenerator {
  constructor(args, opts) {
    // ...
  }

  private async _isConnectorType(connectorType, dataSourceClassName) {
    // ...
  }
}
```